libraries/tock-register-interface/src/macros.rs

```diff
@@ -48,11 +48,79 @@ macro_rules! register_bitmasks {
         $( $crate::register_bitmasks!($valtype, $reg_desc, $(#[$inner])* $field, $offset, $numbits,
                               $values); )*
     };
+    {
+        $valtype:ty, $reg_desc:ident, $(#[$outer:meta])* $field:ident,
+                    $offset:expr, $numbits:expr,
+                    [$( $(#[$inner:meta])* $valname:ident = $value:expr ),+]
+    } => { // this match arm is duplicated below with an allowance for 0 elements in the valname -> value array,
+        // to seperately support the case of zero-variant enums not supporting non-default
+        // representations.
+        #[allow(non_upper_case_globals)]
+        #[allow(unused)]
+        pub const $field: Field<$valtype, $reg_desc> =
+            Field::<$valtype, $reg_desc>::new((1<<($numbits-1))+((1<<($numbits-1))-1), $offset);
```

Comment:
I think I commented on this pattern before, but can't retrieve where. This arithmetic expression is quite unreadable, and appears many times in these macros. Could you add some helper macro or `const fn` that computes it? As I understand it, it a bitmask for `$numbits`, so `bitmask($numbits)` would be more readable here.

Also, an alternative expression would be: `!0 >> (8 * core::mem::size_of::<$valtype>() - $numbits)`, which may be more readable (fewer nested parentheses to parse for the reader).